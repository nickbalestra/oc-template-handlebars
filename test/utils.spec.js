const utils = require('../packages/oc-template-handlebars/utils');

describe('utils module', () => {
  describe('when validating component', () => {
    describe("when component's handlebars version is obsolete", () => {
      const template = {
        compiler: [6, '>= 2.0.0-beta.1'],
        main: function() {
          return 'Hello world!';
        },
        useData: !0
      }; // eslint-disable-line
      const result = utils.validator(template);

      test('should error', () => {
        expect(result.isValid).toBe(false);
        expect(result.error).toBe(
          "The component can't be rendered because it was published with an older OC version"
        );
      });
    });

    describe("when component's handlebars version is supported", () => {
      const template = {
        compiler: [7, '>= 4.0.0'],
        main: function() {
          return 'Hello world!';
        },
        useData: !0
      }; // eslint-disable-line
      const result = utils.validator(template);

      test('should be valid', () => {
        expect(result.isValid).toBe(true);
      });
    });
  });
});
