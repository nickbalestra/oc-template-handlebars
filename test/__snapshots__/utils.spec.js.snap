// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils module should correctly validate template 1`] = `
Object {
  "error": "The component can't be rendered because it was published with an older OC version",
  "isValid": false,
}
`;

exports[`utils module should correctly validate template 2`] = `
Object {
  "isValid": true,
}
`;
